contrib/win32/openssh/OpenSSHUtils.psm1

```diff
@@ -829,4 +829,25 @@ function Enable-Privilege {
     $type[0]::EnablePrivilege($Privilege, $Disable)
 }
 
-Export-ModuleMember -Function Repair-FilePermission, Repair-SshdConfigPermission, Repair-SshdHostKeyPermission, Repair-AuthorizedKeyPermission, Repair-UserKeyPermission, Repair-UserSshConfigPermission, Enable-Privilege, Get-UserAccount, Get-UserSID, Repair-AdministratorsAuthorizedKeysPermission, Repair-ModuliFilePermission, Repair-SSHFolderPermission, Repair-SSHFolderFilePermission, Repair-SSHFolderPrivateKeyPermission
+Function Add-Path {
+    [CmdletBinding(SupportsShouldProcess=$true, ConfirmImpact="High")]
+    param
+    (
+        [parameter(Mandatory=$true)]
+        [string]$FilePath
+    )
+
+    if (Test-Path $FilePath) {
+        $machinePath = (Get-ItemProperty -Path ‘Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment’ -Name PATH).Path
```

Comment:
You can also use the built-in PowerShell drives for registry:
```powershell
Get-Item 'HKLM:\System\CurrentControlSet\Control\Session Manager\Environment'
```

But for getting/setting environment variables, we usually use dotNet APIs rather than directly manipulating the registry:

```powershell
[System.Environment]::GetEnvironmentVariable('PATH', [System.EnvironmentVariableTarget]::Machine)
[System.Environment]::SetEnvironmentVariable('PATH', $newPath, [System.EnvironmentVariableTarget]::Machine)
```

Reply:
Using built-in PowerShell drives, since .NET APIs will Expand Strings, as mentioned in https://github.com/PowerShell/Win32-OpenSSH/issues/1960#issuecomment-1192490353